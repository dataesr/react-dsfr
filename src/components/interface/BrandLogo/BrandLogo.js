import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';
import classNames from 'classnames';

/**
 *
 * @visibleName Bloc-marque
 */
const BrandLogo = ({
  children, href, className, splitCharacter, hrefTitle,
}) => {
  // TODO Better system to split
  const regex = new RegExp(`.{${splitCharacter.toString()}}\\S*\\s+`, 'g');
  const arrayStr = children.replace(regex, '$&@').split(/\s+@/);
  const title = arrayStr.reduce((el, a, i) => el.concat(a, i < arrayStr.length - 1 ? (
    <br
      key={uuidv4()}
    />
  ) : ''), []);
  return (
    <a
      className={classNames('rf-logo', className)}
      href={href}
      title={hrefTitle || children}
    >
      <span className="rf-logo__title">
        {title}
      </span>
    </a>
  );
};

BrandLogo.defaultProps = {
  href: '/',
  hrefTitle: '',
  splitCharacter: 12,
  className: '',
};
BrandLogo.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array,
  ]),
  children: PropTypes.string.isRequired,
  hrefTitle: PropTypes.string,
  href: PropTypes.string,
  /**
  * Number of characters to split title
  */
  splitCharacter: PropTypes.number,
};

export default BrandLogo;
