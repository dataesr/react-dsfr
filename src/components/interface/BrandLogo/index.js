import '@gouvfr/dsfr/dist/css/logo.min.css';

export { default } from './BrandLogo';
