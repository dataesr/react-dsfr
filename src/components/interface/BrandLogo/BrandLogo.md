## Le bloc marque
La structure du bloc-marque est commune à toutes les entités de l’État.

```jsx
import BrandLogo from '.';

<BrandLogo>
    Ministère de l'enseignement supérieur de la recherche et de l'innovation
</BrandLogo>
```
