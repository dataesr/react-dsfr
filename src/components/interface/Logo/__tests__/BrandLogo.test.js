import { render, screen } from '@testing-library/react';
import BrandLogo from '..';

describe('<Logo />', () => {
  it('should render Logo properly', () => {
    render(
      <BrandLogo>
        Ministère de l&apos;enseignement supérieur de la rechercher et de l&apos;innovation
      </BrandLogo>,
    );
    const footer = screen.getByText(/Ministère/);
    expect(footer).toMatchSnapshot();
  });
});
