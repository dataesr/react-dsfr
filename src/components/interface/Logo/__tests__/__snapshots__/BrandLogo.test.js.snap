// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render Logo properly 1`] = `
<span
  class="rf-logo__title"
>
  Ministère de
  <br />
  l'enseignement
  <br />
  supérieur de
  <br />
  la rechercher
  <br />
  et de l'innovation
  
</span>
`;
